File: src/test/java/MaximumSubarrayTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MaximumSubarrayTest {

    private final MaximumSubarray maximumSubarray = new MaximumSubarray();

    @Test
    void testSample() {
        int[] given = new int[]{-2, 1, -3, 4, -1, 2, 1, -5, 4};
        int expected = 6;

        int actual = maximumSubarray.maxSubArray(given);

        assertEquals(expected, actual);
    }

    @Test
    void testOnlyPositive() {
        int[] given = new int[]{1, 2};
        int expected = 3;

        int actual = maximumSubarray.maxSubArray(given);

        assertEquals(expected, actual);
    }

    @Test
    void testOnlyNegative() {
        int[] given = new int[]{-1, -2};
        int expected = -1;

        int actual = maximumSubarray.maxSubArray(given);

        assertEquals(expected, actual);
    }
}

File: src/test/kotlin/MaximumSubarrayTest.kt
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class MaximumSubarrayTest {
    private val maximumSubarray = MaximumSubarray()
    @Test
    fun testSample() {
        val given = intArrayOf(-2, 1, -3, 4, -1, 2, 1, -5, 4)
        val expected = 6
        val actual = maximumSubarray.maxSubArray(given)
        Assertions.assertEquals(expected, actual)
    }

    @Test
    fun testOnlyPositive() {
        val given = intArrayOf(1, 2)
        val expected = 3
        val actual = maximumSubarray.maxSubArray(given)
        Assertions.assertEquals(expected, actual)
    }

    @Test
    fun testOnlyNegative() {
        val given = intArrayOf(-1, -2)
        val expected = -1
        val actual = maximumSubarray.maxSubArray(given)
        Assertions.assertEquals(expected, actual)
    }
}
